source/develop/release-management/features/virt/vm-failover-using-vm-leases.html.md

```diff
@@ -0,0 +1,181 @@
+host when the host becomes unreachable and the host cannot be fenced.
+In the current system the VM state is unknown and the system cannot
+start the vm on one of the available hosts until the host is rebooted
+manually and the administrator mark the host as rebooted manually.
+
+This feature is tracked in bug
+https://bugzilla.redhat.com/804272
+
+## Owner
+
+- Name: [Nir Soffer](https://github.com/nirs)
+- Email: <nsoffer@redhat.com>
+
+
+## Detailed Description
+
+
+### How it works
+
+To enable this feature, the user must create a storage lease for the vm.
+Only VM holding a storage lease can use automatic failover when a host
+or storage becomes unreachable.
+
+When a host is connected to storage sanlock will renew the host lease
+regularly. The system can inspect a host lease status status by using
+`Lease.host_status` api. Lease status should be `LIVE` in this case.
+
+When a VM is holding a lease on storage, the system can inspect the
+lease status using the `Lease.info` api. The lease status should be
+`EXCLUSIVE` when the VM is running.
+
+If a host looses power, it stops renewing the lease.  Inspecting the
+host lease status from another host will show the lease changing state
+to `FAIL` and finally to `DEAD`.
+
+When sanlock cannot renew the host lease, it will terminate all the
+processes holding a lease in that storage. The system can detect this
+situation by inspecting the host lease status. The lease state will
+change to `FAIL` and finally to `DEAD`.
+
+If a VM holding a lease is terminated, sanlock should notice that the
+process holding a lease is not running and release the lease. Inspecting
+the VM lease from another host should show that the VM lease is `FREE`,
+and the host lease status is `LIVE`.
+
+    Note: need to check the details with David
+
+When a host become unreachable, the system will check the host lease
+status and VM leases status on this host via another host in the DC
+connected to the same storage domains.
+
+Once the system has determined that a host lease is `DEAD` or a VM lease
+is `FREE`, it can start the VM on another host. From the user point of
+view, the behavior is the same as if the user rebooted the host and
+marked the host as "rebooted" manually.
+
+
+### Supported failure modes
+
+This feature will allow failover in the following modes:
+
+- Power failure - the host lost power, so no VM is running on the host.
+  Host lease status is `DEAD`.
+
+- Inaccessible host and storage - VM holding a lease on this host will
+  be terminated by sanlock. Host lease status is `DEAD`.
+
+- Inaccessible host, VM terminated - VM holding a lease on this host
+  was terminated for another reason, the VM lease status is `FREE`.
+
+
+### Unsupported failure modes
+
+This feature is not effecting these modes:
+
+- Networking issue - the host is unreachable because of networking issue
+  but the host can still access storage via a another network or via FC.
+  The host lease status will be `LIVE`.
+
```

Comment:
Today you can decide whether you want to fence a host in case of networking issue, based on whether the host maintains its lease or not.
What would be the behavior this time?
Will it be configurable?
Will the fact you're using leases means fencing is disabled as a whole on this DC?
Will the leases be on by default? Can someone disable the use of storage leases?